### DISCUSSION TITLE
How-to: importing map conditional styles from one existing map to another existing map

### DISCUSSION THREAD

**Discussion by Illegit** [freeplane/freeplane#1366] [Help] [answered]
Hi All, I am asking you for any suggestion to import map conditional styles from one existing map to another existing map, Please consider that the receiving map has already its own map conditional styles that I do not want to miss.
Thanks

**Comment by sifran-github** [comment, marked as answer]
1. you have a save and load buttons when you edit the conditions (manage conditional styles for map -> edit any):
![image](https://github.com/freeplane/freeplane/assets/5730355/155c2d43-ae7e-4c8b-a1ec-c9e544f3e289)

2. if you have many conditions that you want to copy, you might want to consider using a text editor to do it manually:
![image](https://github.com/freeplane/freeplane/assets/5730355/b9070c50-a139-41b4-852e-c5e4aa1212a7)
just be careful because you need to make sure that you know what you are doing if you open the .mm file in a text editor....

**Comment by Illegit** [comment]
Thankyou sifran-github, I will use the editor.

**Comment by macmarrum** [comment]
I think Conditions and Map Conditional Styles are different animals, although Conditional Styles do use Conditions.

A Conditional Style consists of 
- Active
- Condition
- Style
- Last (aka Stop)

![manage-conditional-styles-for-map](https://github.com/freeplane/freeplane/assets/7197131/79f4843b-f274-4f79-9296-1c20d0c291c2)

The easiest way to copy the whole bunch is by following a mind map: `Format->Manage Styles->Manage followed and associated mind maps` and using "Copy styles". Then you can "Unfollow" so that the target mind map isn't updated each time the source mind map changes styles.

Alternatively, you can use a script to copy Conditional Style entries, but it won't copy the Styles, so it makes sense only if you already have the Styles in your target mind map or you want to use different ones and set them manually afterwards.

```groovy
def file = new File(/c:\path\to\the\source\mindmap\with\conditional\styles/)
def source = c.mapLoader(file).mindMap
source.conditionalStyles.each { node.mindMap.conditionalStyles.add(it) }
```

**Reply by Illegit** [reply]
Thankyou macmarrum for your explanation
